setup.py

```diff
@@ -91,11 +93,42 @@ def __init__(self, name, sourcedir=''):
 
 class CMakeBuild(build_ext):
     def run(self):
+        # Default cmake to whichever one is first in the path.
+        self.cmake_exe = 'cmake.exe' if platform.system() == "Windows" else 'cmake'
+        # Try to find the python distrubution (pip installed) cmake.
+        found_cmake = ''
         try:
-            out = subprocess.check_output(['cmake', '--version'])
-        except OSError:
-            raise RuntimeError("CMake must be installed to build the following extensions: " +
-                               ", ".join(e.name for e in self.extensions))
+            import cmake as cmake_module
+            imported_cmake_module = True
+        except:
+            imported_cmake_module = False
+        if imported_cmake_module:
+            # Although it probably should not happen, robustly deal with multiple cmakes installed
+            #   into the cmake packages' path by taking the cmake with the shallowest path depth.
+            found_cmake_depth = -1
+            for cmake_path in cmake_module.__path__:
+                for cmake in Path(cmake_path).glob('**/{}'.format(self.cmake_exe)):
+                    cmake_exe = cmake.as_posix()
+                    cmake_depth = len(cmake.parts)
+                    try:
+                        out = subprocess.check_output([cmake_exe, '--version'])
+                        valid_cmake = True
+                    except:
+                        valid_cmake = False
+                    if valid_cmake and (not found_cmake or cmake_depth < found_cmake_depth):
+                        found_cmake = cmake_exe
+                        found_cmake_depth = cmake_depth
```

Comment:
I'm a little concerned that the code seems to be overriding the convention of using the first occurrence in $PATH with using the "shallowest" occurrence in $PATH. The user can reasonably shuffle PATH order but if they have to read the setup.py to figure out how to get it to build that seems non-ideal. The intent of looping through till you find a working one seems reasonable but maybe test the version number of output is >= 2.8.12 (the minimum requirement in the CMakeLists.txt file so that it doesn't pass the setup.py and then blow up in `cmake ..`.

Reply:
> I'm a little concerned that the code seems to be overriding the convention of using the first occurrence in $PATH with using the "shallowest" occurrence in $PATH.

Yeah, that's a good point. Before the last set of build changes you introduced, the build command was not taking the pip-installed cmake. I thought this was even after the addition of the cmake package as a build requirement, but now I can not reproduce this. I tested again and it's indeed taking pip-installed cmake on both platforms without this change.

So, I agree this could potentially override the user's path and additionally seems no longer necessary.

The other change is super minor, so I'll delete this pull request.